test/integration/client-side-encryption/client_side_encryption.prose.test.js

```diff
@@ -25,994 +43,1415 @@ const noop = () => {};
-              expect(awsDatakeyId).to.have.property('sub_type', 4);
-            })
-            .then(() => {
-              return this.client
-                .db(keyVaultDbName)
-                .collection(keyVaultCollName)
-                .find({ _id: awsDatakeyId })
-                .toArray();
-            })
-            .then(results => {
-              expect(results)
-                .to.have.a.lengthOf(1)
-                .and.to.have.nested.property('0.masterKey.provider', 'aws');
-              expect(this.commandStartedEvents.events).to.containSubset([
-                { commandName: 'insert', command: { writeConcern: { w: 'majority' } } }
-              ]);
-            });
-        })
-        .then(() => {
-          // #. Call ``client_encryption.encrypt()`` with the value "hello aws", the algorithm ``AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic``, and the ``key_id`` of ``aws_datakey_id``.
-          //    - Expect the return value to be a BSON binary subtype 6, referred to as ``aws_encrypted``.
-          //    - Use ``client_encrypted`` to insert ``{ _id: "aws", "value": <aws_encrypted> }`` into ``db.coll``.
-          //    - Use ``client_encrypted`` to run a find querying with ``_id`` of "aws" and expect ``value`` to be "hello aws".
-          const coll = this.clientEncrypted.db(dataDbName).collection(dataCollName);
-          return this.clientEncryption
-            .encrypt('hello aws', {
-              algorithm: 'AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic',
-              keyId: awsDatakeyId
-            })
-            .then(value => {
-              awsEncrypted = value;
-              expect(awsEncrypted).to.have.property('sub_type', 6);
-            })
-            .then(() => coll.insertOne({ _id: 'aws', value: awsEncrypted }))
-            .then(() => coll.findOne({ _id: 'aws' }))
-            .then(result => {
-              expect(result).to.have.property('value', 'hello aws');
-            });
-        })
-        .then(() => {
-          // #. Call ``client_encryption.encrypt()`` with the value "hello aws", the algorithm ``AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic``, and the ``key_alt_name`` of ``aws_altname``.
-          //    - Expect the return value to be a BSON binary subtype 6. Expect the value to exactly match the value of ``aws_encrypted``.
-          return this.clientEncryption
-            .encrypt('hello aws', {
-              algorithm: 'AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic',
-              keyId: awsDatakeyId
-            })
-            .then(encrypted => {
-              expect(encrypted).to.deep.equal(awsEncrypted);
-            });
+          const masterKeyExpired = { ...masterKey, endpoint: '127.0.0.1:8000' };
+          const masterKeyInvalidHostname = { ...masterKey, endpoint: '127.0.0.1:8001' };
+
+          it('fails with no tls', metadata, async function () {
+            try {
+              await clientEncryptionNoTls.createDataKey('aws', { masterKey });
```

Comment:
I'll stop commenteing but all these catches need to fail in the success case

Reply:
Added.